proxy/http/HttpConfig.cc

```diff
@@ -65,16 +67,18 @@ class HttpConfigCont : public Continuation
 /// Data item for enumerated type config value.
 template <typename T> struct ConfigEnumPair {
   T _value;
-  const char *_key;
+  ts::StringView _key;
+
+  ConfigEnumPair(T v, char const *k) : _value(v), _key(k) {}
 };
 
 /// Convert a string to an enumeration value.
 /// @a n is the number of entries in the list.
 /// @return @c true if the string is found, @c false if not found.
 /// If found @a value is set to the corresponding value in @a list.
-template <typename T, unsigned N>
+template <typename E, unsigned N>
 static bool
-http_config_enum_search(const char *key, const ConfigEnumPair<T> (&list)[N], MgmtByte &value)
+http_config_enum_search(ts::StringView key, const ConfigEnumPair<E> (&list)[N], MgmtByte &value)
```

Comment:
It's experimental at this time. It's an expansion of what's being done for session sharing and so for now I left it in the same place.  If it works out then yes, it should be moved to LibRecords which in turn should be updated to directly support enums.